### PR TITLE
Add upgrade notes and docs for JS events

### LINKED ISSUE #12591: Add upgrade notes and docs for JS events
### Improvement description

Follow up to https://github.com/pimcore/pimcore/pull/12566

- [x] Missing hint in upgrade notes that the old plugin broker got disabled
- [x] Docs should be updated to use the new events and old plugin broker got disabled
- [x] If possible we should issue a `console.warn()` if someone registers a listener the old way
- [x] add migration notes

### PR DESCRIPTION
**Pull request by robertSt7**
resolves #12591

### CODE REVIEW

File: bundles/AdminBundle/Resources/public/js/pimcore/plugin/broker.js [hunk 1]
```diff
@@ -37,6 +38,11 @@ pimcore.plugin.broker = {
 
     executePlugin: function (plugin, event, params) {
         if (typeof plugin[event] == "function") {
+            const warnText = `Plugins are depreacted. Please use event listener - ${plugin.getClassName()}::${event}`;
```
Comment (mcop1) [inline bundles/AdminBundle/Resources/public/js/pimcore/plugin/broker.js]:
```suggestion
            const warnText = `Plugins are deprecated. Please use event listener - ${plugin.getClassName()}::${event}`;
```

File: doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md [hunk 2]
```diff
@@ -154,98 +131,65 @@ The only difference between above exceptions is in how the error message is disp
...
 });
-
-var MyTestBundlePlugin = new pimcore.plugin.MyTestBundle();
 ```
 
-## I18n texts for plugins
+## I18n texts for js
 
-Pimcore supports i18n for plugin UI extensions. Fist see the [i18n section for bundles](./README.md) how to prepare the data 
+Pimcore supports i18n for UI extensions. Fist see the [i18n section for bundles](./README.md) how to prepare the data
```
Comment (mcop1) [inline doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md]:
```suggestion
Pimcore supports i18n for UI extensions. First see the [i18n section for bundles](./README.md) how to prepare the data 
```

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/README.md [hunk 3]
```diff
@@ -24,4 +24,5 @@ use: `bin/console doctrine:migrations:migrate --prefix=Pimcore\\Bundle\\CoreBund
 
 ## Upgrading to Version X from earlier Versions
```
Comment (mcop1) [inline doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/README.md]:
```suggestion
## Upgrading from earlier Versions
```

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/README.md [hunk 4]
```diff
@@ -22,6 +22,7 @@ If you do not want to run Pimcore migrations automatically please remove `"Pimco
 To run core migrations manually (e.g. when using composer install), 
 use: `bin/console doctrine:migrations:migrate --prefix=Pimcore\\Bundle\\CoreBundle`
 
-## Upgrading to Version X from earlier Versions
+## Upgrading from earlier Versions
 - [Upgrade from Version 6 to Version X](./10_V6_to_V10.md)
+- [Upgrade from Version 10 to Version XI](./11_V10_to_V11.md)
```
Comment (jdreesen) [inline doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/README.md]: Why not:

```suggestion
- [Upgrade from Version X to Version XI](./11_V10_to_V11.md)
```

File: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 5]
```diff
@@ -21,9 +21,11 @@
   Implement an [EventListener](https://github.com/pimcore/pimcore/blob/10.x/bundles/EcommerceFrameworkBundle/EventListener/SessionBagListener.php) to register a session bag before the session is started.
 - [Ecommerce][PricingManager] Token condition is deprecated and will be removed in Pimcore 11.
 - Parameter `pimcore.admin.session.attribute_bags` is deprecated and will be removed in Pimcore 11.
+- [JS Events/Disabled] The plugin broker for fire events has been disabled.
```
Comment (brusch) [inline doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:24]: Here we should include a link to the event listener docs.

File: doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md [hunk 6]
```diff
@@ -56,40 +56,22 @@ class PimcoreAdminListener
...
-var samplePlugin = new pimcore.plugin.sample();
 ```
 
 ## JavaScript UI Events
 
-The broker then will notify each plugin upon the events described below. For registering to these events just add a 
-corresponding method to the javascript plugin class. 
+For registering events just add a listener with some of the events from [events.js](https://github.com/pimcore/pimcore/blob/10.x/bundles/AdminBundle/Resources/public/js/pimcore/events.js). 
 
 | Name                                 | Description                                                                                                     | Notes    |
```
Comment (brusch) [inline doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md]: I think we could remove the table from here and just move the event descriptions over to the JS file. Then we have just one place to maintain the stuff and we are doing the same for the PHP events.

File: doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md [hunk 7]
```diff
@@ -154,98 +131,65 @@ The only difference between above exceptions is in how the error message is disp
...
-        });
-        pimcore.layout.toolbar.prototype.mdsMenu = this.menu;
-    },
- 
-    pimcoreReady: function (params, broker) {
-        var toolbar = pimcore.globalmanager.get("layout_toolbar");
-        this.navEl.on("mousedown", toolbar.showSubMenu.bind(toolbar.mdsMenu));
-        pimcore.plugin.broker.fireEvent("mdsMenuReady", toolbar.mdsMenu);
-    }
+this.navEl = Ext.get('pimcore_menu_search').insertSibling('<li id="pimcore_menu_mds" data-menu-tooltip="mds extension" class="pimcore_menu_item pimcore_menu_needs_children">mds extension</li>', 'after');
```
Comment (brusch) [inline doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md]: I think we should use a local variables here `var` should be sufficient here, right?

**Review by brusch — CHANGES_REQUESTED** [review]

**Pull request merged**